docs/macros.py

```diff
@@ -0,0 +1,76 @@
+import os
+
+def declare_variables(variables, macro):
+    @macro
+    def code_snippet(filename: str, snippet: str = "", language: str = "cpp"):
+        """
+        Load code from a file and save as a preformatted code block.
```

Comment:
There's no explanation related to that anywhere. But this is basically for generating the documentation code snippets from the examples. Something we definitely need in the next iterations of this documentation.